src/GafferVDB/LevelSetToMesh.cpp

```diff
@@ -271,19 +255,23 @@ IECore::ConstObjectPtr LevelSetToMesh::computeProcessedObject( const ScenePath &
 	return volumeToMesh( grid, isoValuePlug()->getValue(), adaptivityPlug()->getValue() );
 }
 
-bool LevelSetToMesh::processesBound() const
+bool LevelSetToMesh::affectsProcessedObjectBound( const Gaffer::Plug *input ) const
 {
-	return true;
+	return input == inPlug()->boundPlug() || input == isoValuePlug();
 }
 
-void LevelSetToMesh::hashProcessedBound( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+void LevelSetToMesh::hashProcessedObjectBound( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
 {
-	SceneElementProcessor::hashProcessedBound( path, context, h );
+	inPlug()->boundPlug()->hash( h );
 	isoValuePlug()->hash( h );
 }
 
-Imath::Box3f LevelSetToMesh::computeProcessedBound( const ScenePath &path, const Gaffer::Context *context, const Imath::Box3f &inputBound ) const
+Imath::Box3f LevelSetToMesh::computeProcessedObjectBound( const ScenePath &path, const Gaffer::Context *context ) const
 {
+	/// \todo `in.bound` includes the child bounds. Ideally we
+	/// would have a separate `in.objectBound` with just the bounds
+	/// of the input object.
+	const Box3f inputBound = inPlug()->boundPlug()->getValue();
 	const V3f offset( isoValuePlug()->getValue() );
 	return Box3f( inputBound.min - offset, inputBound.max + offset );
```

Comment:
So are we saying that LevelSetToMesh shouldn't even be a Deformer then?

Reply:
No, wait, I think I get you now. The `isoValue` is arbitrary, and we don't know where it will occur in the grid, so the offset approach we're taking is clearly wrong. But it _can_ change the bound, so we do want to be a Deformer still. But we should use the default implementation for bounds, so we get accurate bounds for the new grid by default. And we can turn off `adjustBounds` if we want to fall back to using the big-enough-but-possibly-too-big input bound.  Does that sound right?

Reply:
That sounds correct.  I think I would prefer adjustBounds to default off in this case, since the unadjusted bound is always going to be correct anyway ( it may just be overly conservative ).  But I guess there are case where having a more accurate bound is nice, and I suppose for consistency with other nodes, you may want to default it on - that's kinda subjective ( I guess it's not that slow comparatively to scan a million vertices looking for bounds after generating them ).  But the important thing is when we're not doing the work computing tight bounds, we should pass through the input bound, rather than assuming it's a level set and using adjusted bounds that could be incorrectly small if the input data isn't a level set.

Reply:
OK, I've implemented this in https://github.com/GafferHQ/gaffer/pull/4260, including defaulting the `adjustBounds` off.